core/designsystem/src/main/kotlin/com/dhc/designsystem/calendar/model/DhcCalendarDayUiModel.kt

```diff
@@ -0,0 +1,73 @@
+    val textColor: Color @Composable get
+
+    data object NoCurrentMonth: DhcCalendarDayUiModel {
+        override val backgroundColor @Composable get() = SurfaceColor.neutral600
+        override val borderColor @Composable get() = null
+        override val textStyle = DhcTypoTokens.Body3
+        override val textColor @Composable get() = SurfaceColor.neutral400
+    }
+
+    data object Today: DhcCalendarDayUiModel {
+        override val backgroundColor @Composable get() = LocalDhcColors.current.background.backgroundBadgePrimary
+        override val borderColor @Composable get() = LocalDhcColors.current.text.textHighLightsPrimary
+        override val textStyle = DhcTypoTokens.TitleH5
+        override val textColor @Composable get() = LocalDhcColors.current.text.textMain
+    }
+
+    data object DidNotMission: DhcCalendarDayUiModel {
+        override val backgroundColor @Composable get() = SurfaceColor.neutral600
+        override val borderColor @Composable get() = null
+        override val textStyle = DhcTypoTokens.Body3
+        override val textColor @Composable get() = LocalDhcColors.current.text.textBodyPrimary
+    }
+
+    data object DidOneMission: DhcCalendarDayUiModel {
+        override val backgroundColor @Composable get() = AccentColor.violet400.copy(alpha = 0.35f)
+        override val borderColor @Composable get() = null
+        override val textStyle = DhcTypoTokens.Body3
+        override val textColor @Composable get() = LocalDhcColors.current.text.textBodyPrimary
+    }
+
+    data object DidTwoMission: DhcCalendarDayUiModel {
+        override val backgroundColor @Composable get() = AccentColor.violet400.copy(alpha = 0.7f)
+        override val borderColor @Composable get() = null
+        override val textStyle = DhcTypoTokens.Body3
+        override val textColor @Composable get() = LocalDhcColors.current.text.textBodyPrimary
+    }
+
+    data object DidMoreMission: DhcCalendarDayUiModel {
+        override val backgroundColor @Composable get() = LocalDhcColors.current.text.textHighLightsPrimary
+        override val borderColor @Composable get() = null
+        override val textStyle = DhcTypoTokens.Body3
+        override val textColor @Composable get() = LocalDhcColors.current.text.textBodyPrimary
+    }
+
+    companion object {
+        fun from(day: LocalDate, monthData: DhcCalendarMonthData): DhcCalendarDayUiModel {
+            return when {
+                day.month != monthData.yearMonth.month -> NoCurrentMonth
+                day == LocalDate.now() -> Today
+                monthData.data[day.dayOfMonth]?.finishedMissionCount == 0 -> DidNotMission
+                monthData.data[day.dayOfMonth]?.finishedMissionCount == 1 -> DidOneMission
+                monthData.data[day.dayOfMonth]?.finishedMissionCount == 2 -> DidTwoMission
+                monthData.data[day.dayOfMonth]?.finishedMissionCount == 3 -> DidMoreMission
+                else -> DidNotMission
+            }
+        }
```

Comment:
```suggestion
        fun from(day: LocalDate, monthData: DhcCalendarMonthData): DhcCalendarDayUiModel {
            val finishedMissionCount = monthData.data[day.dayOfMonth]?.finishedMissionCount
            return when {
                day.month != monthData.yearMonth.month -> NoCurrentMonth
                day == LocalDate.now() -> Today
                finishedMissionCount == 0 -> DidNotMission
                finishedMissionCount == 1 -> DidOneMission
                finishedMissionCount == 2 -> DidTwoMission
                finishedMissionCount == 3 -> DidMoreMission
                else -> DidNotMission
            }
        }
```
P2 : when 문을 타고 탈 수록 `monthData.data[day.dayOfMonth]?.finishedMissionCount` 가 계속 실행 될 것 같아서 맨 첫 단계에서 미리 값을 뽑아놓고 시작하는건 어떨까?

Reply:
@JeonK1 
너무 좋은데 ????